api/src/test/java/org/pmiops/workbench/api/DataDictionaryTest.java

```diff
@@ -0,0 +1,156 @@
+
+  @Autowired BigQueryService bigQueryService;
+  @Autowired CohortDao cohortDao;
+  @Autowired ConceptDao conceptDao;
+  @Autowired ConceptSetDao conceptSetDao;
+  @Autowired DataSetDao dataSetDao;
+  @Autowired DataSetMapper dataSetMapper;
+  @Autowired DataSetService dataSetService;
+  @Autowired FireCloudService fireCloudService;
+  @Autowired NotebooksService notebooksService;
+  @Autowired WorkspaceService workspaceService;
+  @Mock Provider<User> userProvider;
+
+  @Autowired CdrVersionDao cdrVersionDao;
+  @Autowired DataDictionaryEntryDao dataDictionaryEntryDao;
+
+  private static final Instant NOW = Instant.now();
+  private static final FakeClock CLOCK = new FakeClock(NOW, ZoneId.systemDefault());
+  private static User currentUser;
+
+  @TestConfiguration
+  @Import({DataSetMapperImpl.class})
+  @MockBean({
+    BigQueryService.class,
+    CohortDao.class,
+    ConceptDao.class,
+    ConceptSetDao.class,
+    DataSetDao.class,
+    DataSetService.class,
+    FireCloudService.class,
+    NotebooksService.class,
+    WorkspaceService.class
+  })
+  static class Configuration {
+    @Bean
+    Clock clock() {
+      return CLOCK;
+    }
+
+    @Bean
+    @Scope("prototype")
+    User user() {
+      return currentUser;
+    }
+  }
+
+  private DataSetController dataSetController;
+
+  @Before
+  public void setUp() {
+    dataSetController =
+        new DataSetController(
+            bigQueryService,
+            CLOCK,
+            cdrVersionDao,
+            cohortDao,
+            conceptDao,
+            conceptSetDao,
+            dataDictionaryEntryDao,
+            dataSetDao,
+            dataSetMapper,
+            dataSetService,
+            fireCloudService,
+            notebooksService,
+            userProvider,
+            workspaceService);
+  }
+
+  @Test
+  public void testGetDataDictionaryEntry() {
+    final Domain domain = Domain.DRUG;
+    final String domainValue = "FIELD NAME / DOMAIN VALUE";
+
+    CdrVersion cdrVersion = new CdrVersion();
+    cdrVersionDao.save(cdrVersion);
+
+    DataDictionaryEntry dataDictionaryEntry = new DataDictionaryEntry();
+    dataDictionaryEntry.setCdrVersion(cdrVersion);
+    dataDictionaryEntry.setDefinedTime(new Timestamp(CLOCK.millis()));
+    dataDictionaryEntry.setRelevantOmopTable(ConceptSetDao.DOMAIN_TO_TABLE_NAME.get(domain));
+    dataDictionaryEntry.setFieldName(domainValue);
+    dataDictionaryEntry.setOmopCdmStandardOrCustomField("A");
+    dataDictionaryEntry.setDescription("B");
+    dataDictionaryEntry.setFieldType("C");
+    dataDictionaryEntry.setDataProvenance("D");
+    dataDictionaryEntry.setSourcePpiModule("E");
+    dataDictionaryEntry.setTransformedByRegisteredTierPrivacyMethods(true);
+
+    dataDictionaryEntryDao.save(dataDictionaryEntry);
+
+    org.pmiops.workbench.model.DataDictionaryEntry response =
+        dataSetController
+            .getDataDictionaryEntry(cdrVersion.getCdrVersionId(), domain.toString(), domainValue)
```

Comment:
nit: this isn't really exercising the matching functionality since there's only one entry in the database, may want to insert another entry